### PR TITLE
[21.05] Add support for RDS files and improvements for RData

### PR DESCRIPTION
**Pull request by bernt-matthias**
Irrespective of the controversy on the pros and cons of Rdata we should do it correctly. Currently Galaxy only has `RData` (i.e. the result of R's `save`/`save.img` function which saves multiple objects or the complete environment), but often its RDS data (i.e. the result of R's `saveRDS` function which saves a single object).

See also https://cran.r-project.org/doc/manuals/r-patched/R-ints.html#Serialization-Formats

For RData version 3 RData serialization is supported (default since R 3.5.) and version metadata element has been added.

With the metadata elements it might be possible to setup dataset filters, but I'm not sure if `RDS`'s `rversion` and `minrversion` are useful. So we might also remove them.

I have chosen to target 21.05 because the dada datatypes (which are actually rds) have been introduced in this release. Also soe IUC tools use rds (but abuse rdata for this).

TODOs

- [x] upload test data
- [x] fix metadata setting .. seems that Galaxy unzips the data when stored in the database
- [x] maybe switch gzipped and normal reading since R's default is gzipped output
  - I guess the improvement would be marginal
- [x] fix some subclassed datatypes, e.g. `dada*`
- [x] remove debug statements

## How to test the changes?
(Select all options that apply)
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

**Comment by nsoranzo** [comment]
> 
> 
> Seems that only 3 checks ran.

Closing/reopening did the trick.

File: lib/galaxy/datatypes/binary.py [hunk 1]
```diff
@@ -2308,22 +2308,168 @@ def display_peek(self, dataset):
...
+
     def sniff(self, filename):
-        rdata_header = b'RDX2\nX\n'
+        # try to read plain file
         try:
-            header = open(filename, 'rb').read(7)
-            if header == rdata_header:
-                return True
+            with open(filename, 'rb') as fh:
+                self._sniff_rdata(fh)
```
Comment (mvdbeek) [inline lib/galaxy/datatypes/binary.py]: Could you use the file_prefix.contents_header_bytes for this ? That should work for the compressed and uncompressed case

Comment (bernt-matthias) [reply lib/galaxy/datatypes/binary.py]: Good idea. But I thing its even better to use `compression_utils.get_fileobj_raw` directly. If I understand the code correctly `file_prefix.contents_header_bytes` contains per default the 1MB which seems wasteful here.

Comment (mvdbeek) [reply lib/galaxy/datatypes/binary.py]: It's loaded anyway, so that should be better

Comment (bernt-matthias) [reply lib/galaxy/datatypes/binary.py]: > It's loaded anyway, so that should be better

Isn't the prefix only loaded if I add `@build_sniff_from_prefix`?

Comment (mvdbeek) [reply lib/galaxy/datatypes/binary.py]: ~~For that datatype, but it'll be there for all others that are checked~~

Comment (mvdbeek) [reply lib/galaxy/datatypes/binary.py]: That's not actually how it works, the file prefix should always be loaded, if it's used or not -> https://github.com/mvdbeek/galaxy/blob/ff9146f06df7dc899b38e621852a57bffa265a38/lib/galaxy/datatypes/sniff.py#L462

**Review by mvdbeek — APPROVED** [review]

**Comment by bernt-matthias** [comment]
Thanks @mvdbeek for finishing this

**Pull request merged**